src/services/latex.js

```diff
@@ -77,6 +77,10 @@ Controller.open(function(_, super_) {
   _.exportLatex = function() {
     return this.root.latex().replace(/(\\[a-z]+) (?![a-z])/ig,'$1');
   };
+
+  optionProcessors.maxDepth = function(depth) {
+    return (typeof depth === 'number') ? depth : undefined;
```

Comment:
If the default in options for maxDepth was `Number.MAX_SAFE_INTEGER || Number.MAX_VALUE` then the undefined checks used elsewhere are unnecessary.  I mean we could choose another arbitrarily large number, but those are built into JavaScript.

Reply:
Hi @kschaefe, the undefined checks are really there for performance reasons. If maxDepth defaults to some large number, then we'll be doing an unnecessary breadth-first search on every insertion.

Reply:
Gotcha.  Yes, a fast-exit from the function is preferable to a search every time.  It totally worth that complexity then.